lxd/device/nic_routed.go

```diff
@@ -110,6 +112,33 @@ func (d *nicRouted) validateConfig(instConf instance.ConfigReader) error {
 		}
 	}
 
+	for _, keyPrefix := range []string{"ipv4", "ipv6"} {
+		addr, ok := d.config[fmt.Sprintf("%s.address", keyPrefix)]
+		if ok {
+			p, err := ping.NewPinger(addr)
+			if err != nil {
+				return errors.Wrapf(err, "Could not determine if the %s address %q is in use", keyPrefix, addr)
+			}
+
+			inUse := true
+			p.SetPrivileged(true)
+			p.Timeout = 100 * time.Millisecond
+			p.Count = 1
+			p.OnFinish = func(stats *ping.Statistics) {
+				inUse = stats.PacketsRecv != 0
+			}
+
+			err = p.Run()
+			if err != nil {
```

Comment:
If we get `sendto: network is unreachable` we should not treat it as an error and treat it as if the address isn't in use.

Reply:
It was getting quite convoluted to check for `network is unreachable`. I also came across a `no such host` error which we would need to also allow. 

I think that we can reasonably expect that if there are problems pinging the address then the address is likely available so I've just continued the loop in order to check ipv6.

I've also tested this validation for IPv4 and IPv6 in the `container_devices_nic_routed` test.